examples/c/5_series/gnss_ins/5_series_gnss_ins_example.c

```diff
@@ -1,354 +1,1106 @@
-    //
+    mip_interface_register_extractor(
+        &device,
+        &filter_data_handlers[2],
+        MIP_FILTER_DATA_DESC_SET,                        // Data descriptor set
+        MIP_DATA_DESC_FILTER_POS_LLH,                    // Data field descriptor set
+        extract_mip_filter_position_llh_data_from_field, // Callback
+        &filter_position_llh                             // Data field out
+    );
 
-    if(mip_3dm_default_device_settings(&device) != MIP_ACK_OK)
-        exit_gracefully("ERROR: Could not load default device settings!");
+    mip_interface_register_extractor(
+        &device,
+        &filter_data_handlers[3],
+        MIP_FILTER_DATA_DESC_SET,                        // Data descriptor set
+        MIP_DATA_DESC_FILTER_VEL_NED,                    // Data field descriptor set
+        extract_mip_filter_velocity_ned_data_from_field, // Callback
+        &filter_velocity_ned                             // Data field out
+    );
 
+    mip_interface_register_extractor(
+        &device,
+        &filter_data_handlers[4],
+        MIP_FILTER_DATA_DESC_SET,                        // Data descriptor set
+        MIP_DATA_DESC_FILTER_ATT_EULER_ANGLES,           // Data field descriptor set
+        extract_mip_filter_euler_angles_data_from_field, // Callback
+        &filter_euler_angles                             // Data field out
+    );
 
-    //
-    //Setup Sensor data format to 100 Hz
-    //
+    // Resume the device
+    // Note: Since the device was idled for configuration, it needs to be resumed to output the data streams
+    MICROSTRAIN_LOG_INFO("Resuming the device.\n");
+    cmd_result = mip_base_resume(&device);
 
-    uint16_t sensor_base_rate;
+    if (!mip_cmd_result_is_ack(cmd_result))
+    {
+        command_failure_terminate(&device, cmd_result, "Could not resume the device!\n");
+    }
 
-    //Note: Querying the device base rate is only one way to calculate the descriptor decimation.
-    //We could have also set it directly with information from the datasheet (shown in GNSS setup).
+    MICROSTRAIN_LOG_INFO("Sensor is configured... waiting for a valid filter solution.\n");
 
-    if(mip_3dm_imu_get_base_rate(&device, &sensor_base_rate) != MIP_ACK_OK)
-         exit_gracefully("ERROR: Could not get sensor base rate format!");
+    mip_gnss_fix_info_data_fix_type current_fix_type = gnss_fix_info.fix_type;
+    mip_filter_mode                 current_state    = filter_status.filter_state;
 
-    const uint16_t sensor_sample_rate = 100; // Hz
-    const uint16_t sensor_decimation = sensor_base_rate / sensor_sample_rate;
+    // Wait for the device to initialize
+    while (filter_status.filter_state < MIP_FILTER_MODE_GX5_RUN_SOLUTION_VALID)
+    {
+        // Update the device state
+        // Note: This will update the device callbacks to trigger the filter state change
+        mip_interface_update(
+            &device,
+            0,    // Time to wait
```

Comment:
wait time should be nonzero so it allows the thread to sleep while waiting for data. 10 to 100ms should be sufficient. Zero would be used if other processing needs to happen without stalling while waiting for data.